src/machines/java/com/enderio/machines/common/recipe/PaintingRecipe.java

```diff
@@ -0,0 +1,150 @@
+package com.enderio.machines.common.recipe;
+
+import com.enderio.EnderIO;
+import com.enderio.core.common.recipes.OutputStack;
+import com.enderio.machines.common.blockentity.PaintingMachineBlockEntity;
+import com.enderio.machines.common.init.MachineRecipes;
+import com.google.gson.JsonObject;
+import net.minecraft.ResourceLocationException;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraft.world.item.crafting.RecipeSerializer;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.wrapper.RecipeWrapper;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class PaintingRecipe implements MachineRecipe<PaintingRecipe.Container> {
+    private final ResourceLocation id;
+    private final Ingredient input;
+    private final Item output;
+
+    public PaintingRecipe(ResourceLocation id, Ingredient input, Item output) {
+        this.id = id;
+        this.input = input;
+        this.output = output;
+    }
+
+    @Override
+    public boolean matches(Container container, Level pLevel) {
+        return input.test(PaintingMachineBlockEntity.INPUT.getItemStack(container)) && !PaintingMachineBlockEntity.PAINT.getItemStack(container).isEmpty();
+    }
+
+    @Override
+    public int getEnergyCost(Container container) {
+        return 2000;
```

Comment:
Could we maybe drive this from config files?

Reply:
I think the config file would become cluttered easily if every machine got an entry

Reply:
That's fair, but its still a really nice way of being able to configure everything, especially when it comes to balance; means we don't have to dig into a massive number of files to find everything